From 708f31719e9cb9390fa13f433ef8d3e4441e7838 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Thu, 13 Feb 2020 18:00:56 +0200
Subject: [PATCH] [SW-0] goya2: fix return value of security_get_key_hash()

In case no valid keys were found, return 0 instead of
-1 to avoid cases where signed values are handled
differently.

Change-Id: I080293c71564908b86f4e722f31ad93b75c1eb6f
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 5c4116e..de43106 100644
--- a/common/security.c
+++ b/common/security.c
@@ -478,7 +478,7 @@
 	int valid_key_offset = security_get_first_valid_key(customer);
 
 	if (valid_key_offset < 0)
-		return valid_key_offset;
+		return 0;
 
 	valid_key_base = (valid_key_offset * SECURITY_BYTE_PER_DWORD *
 			  SECURITY_DWORD_PER_KEY) + SECURITY_EFUSE_KEYS_OFFSET;
